cpp/include/cudf/dictionary/dictionary_factories.hpp

```diff
@@ -69,8 +71,6 @@ std::unique_ptr<column> make_dictionary_column( column_view const& keys_column,
  *
  * The indices values must be in the range [0,keys_column.size()).
  *
- * The null_mask and null count for the output column are copied from the indices column.
```

Comment:
It's potentially confusing what the difference is between this factory and the one above.

I'd change the brief to say that this factory doesn't make a copy and instead takes ownership of the columns.

```
 * @brief Construct a dictionary column by taking ownership of the provided keys
 * and indices columns. 
```